### ISSUE TITLE
Feature request: littlefs for ESP32

### ISSUE THREAD

**Issue by Jason2866** [maxgerhardt/pio-esp32-esp8266-filesystem-downloader#2]
Great addon! It would be great if you add littlefs for ESP32. We use it for ESP32 already. See [Tasmota](https://github.com/arendst/Tasmota)

**Comment by wimmatthijs** [comment]
Also kindly requesting this feature :)

**Comment by 3gyptian** [comment]
ditto!

**Comment by Munger** [comment]
Has there been any update on this feature?

**Comment by Endorfin35** [comment]
great... any plans for esp32 and littlefs?

**Comment by Jason2866** [comment]
In the meantime we did a workaround. To unpack files correctly mklittlefs needs to be compiled with a different setting. The updated mklittlefs is in pio registry https://registry.platformio.org/tools/tasmota/tool-mklittlefs
and the modified script von Max is [here](https://github.com/arendst/Tasmota/blob/development/pio-tools/custom_target.py)

**Comment by maxgerhardt** [comment]
Yeah I'll try and update this repo's scripts today so that all current possible filesystems (SPIFFS, LittleFS, FatFS) are supported properly. Shouldn't be that much of a deal.

**Comment by Jason2866** [comment]
@maxgerhardt We updated our script (based on yours) to read the LittleFS addresses from the device itself (Tasmota can change partitions on the fly) to make sure the are always correct. No more partition.csv is needed/used.
https://github.com/arendst/Tasmota/blob/development/pio-tools/custom_target.py
